### PR TITLE
Examples

### PR DESCRIPTION
**Pull request by Tobias-Dwyer**
I added simple examples to all of the shape classes. 

## Description

I added examples of each shape object in context of the code. So lets say that you make a polygon object and then get the inscribed circle of the polygon. Well that circle is now it's own shape object with it's own properties and I try to make that clear in the code. 

## Motivation and Context

There were no examples yet. 

## Types of Changes

- [x] Documentation update
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/euclid/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/euclid/blob/master/ContributorAgreement.md).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/euclid/blob/master/ChangeLog.txt) and the [credits](https://github.com/glotzerlab/euclid/blob/master/doc/source/credits.rst).

### CODE REVIEW

File: coxeter/shape_classes/polyhedron.py [hunk 1]
```diff
@@ -60,6 +60,65 @@ class Polyhedron(Shape3D):
             Whether or not the faces of the polyhedron are all convex.
             This is used to determine whether certain operations like
             coplanar face merging are allowed (Default value: False).
+    Example::
+        >>> cube  = coxeter.shape_classes.ConvexPolyhedron([[1,1,1],
+        [1,-1,1],[1,1,-1],[1,-1,-1],[-1,1,1],
+        [-1,-1,1],[-1,1,-1],[-1,-1,-1]])
+        >>> cube  = coxeter.shape_classes
+        .Polyhedron(vertices=cube.vertices,faces=cube.faces)
```
Comment (vyasr) [inline coxeter/shape_classes/polyhedron.py:68]: Is this an actual line break? If so, I'm surprised that this doesn't lead to a syntax error, I guess as long as you don't have a `>>>` beginning the line doctest assumes it's a continuation of the previous one?

**Review by vyasr — APPROVED** [review]
@Tobias-Dwyer thanks, this looks great! I have one question, once we've talked through that I think we're good to merge this.

File: coxeter/shape_classes/polyhedron.py (hunk 1 above)
Comment (Tobias-Dwyer) [reply coxeter/shape_classes/polyhedron.py:68]: Nope, not an actual line break, alas this was one of the lines that I had to shorten for syntax reasons. Would be open to a better way to write it.

Comment (vyasr) [reply coxeter/shape_classes/polyhedron.py:68]: Also there's an extra space before the equal sign ;)

File: coxeter/shape_classes/polyhedron.py [hunk 2]
```diff
@@ -60,6 +60,65 @@ class Polyhedron(Shape3D):
             Whether or not the faces of the polyhedron are all convex.
             This is used to determine whether certain operations like
             coplanar face merging are allowed (Default value: False).
+    Example::
+        >>> cube  = coxeter.shape_classes.ConvexPolyhedron([[1,1,1],
+        [1,-1,1],[1,1,-1],[1,-1,-1],[-1,1,1],
+        [-1,-1,1],[-1,1,-1],[-1,-1,-1]])
+        >>> cube  = coxeter.shape_classes
```
Comment (vyasr) [inline coxeter/shape_classes/polyhedron.py:67]:
```suggestion
        >>> cube = coxeter.shape_classes
```

File: coxeter/shape_classes/polyhedron.py (hunk 1 above)
Comment (vyasr) [reply coxeter/shape_classes/polyhedron.py]: I fixed the space and will merge once tests pass. This looks fine to me.

Comment (Tobias-Dwyer) [reply coxeter/shape_classes/polyhedron.py]: eyo thanks! :D

**Pull request merged**